astropy/table/tests/test_table.py

```diff
@@ -1088,6 +1091,32 @@ def test_convert_numpy_array(self, table_types):
         with pytest.raises(ValueError):
             np_data = np.array(d, dtype=[(str('c'), 'i8'), (str('d'), 'i8')])
 
+    def test_byteswap_fits_array(self, table_types):
```

Comment:
It's worth adding in the comment that FITS tables are guaranteed to be big-endian.

This test doesn't actually cover the case of converting from little to big if run on a big-endian machine.  Maybe a lower-level test like:
```
little = sys.byteorder == 'little'
native = '<' if little else '>'
non_native = '>' if little else '<'
for order in (native, non_native):
    t = Table([Column([1,2], dtype=order + 'f', name='a')])
    tnp = t.as_array()
    assert tnp['a'].dtype.byteorder == native
    tnp = t.as_array(keep_byteorder=True)
    assert tnp['a'].dtype.byteorder == order
```